compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2273,7 +2317,8 @@ class Typer extends Namer
     }
 
     def tryImplicit(fallBack: => Tree) =
-      tryInsertImplicitOnQualifier(tree, pt.withContext(ctx), locked).getOrElse(fallBack)
+      tryInsertImplicitOnQualifier(tree, pt.withContext(ctx), locked)
+        .getOrElse(tryNew(tpd)(tree, pt, fallBack))
```

Comment:
I'm a little unclear on why we need to `tryNew` in two places, could this be explained somewhere ?